sdk/spring-sdk/src/main/scala/kalix/springsdk/impl/ViewDescriptorFactory.scala

```diff
@@ -176,51 +179,76 @@ private[impl] object ViewDescriptorFactory extends ComponentDescriptorFactory {
       component: Class[_],
       tableType: Class[_],
       tableName: String,
-      tableProtoMessageName: String) = {
-    var previousEntityClass: Option[Class[_]] = None
-
-    import ReflectionUtils.methodOrdering
-
-    def invalidComponentException(method: Method, extraMessage: String) = {
-      throw InvalidComponentException(
-        s"Method [${method.getName}] annotated with '@Subscribe.EventSourcedEntity' should either receive " +
-        "a single parameter of one of the event types or " +
-        s"two ordered parameters  of type [${tableType.getName}] and an event type. $extraMessage")
+      tableProtoMessageName: String): Seq[KalixMethod] = {
+
+    def getMethodsWithSubscription(component: Class[_]): Seq[Method] = {
+      val methodsTypeLevel: Array[Method] =
+        if (hasEventSourcedEntitySubscription(component))
+          component.getMethods
+            .filter(hasUpdateEffectOutput)
+        else Array.empty
+      val methodsMethodLevel = component.getMethods
+        .filter(hasEventSourcedEntitySubscription)
+
+      if (!methodsTypeLevel.isEmpty && !methodsMethodLevel.isEmpty) {
+        throw InvalidComponentException("Can't annotate the class and individual methods at the same time.")
+      }
+      (methodsTypeLevel ++ validateSameType(methodsMethodLevel, tableType)).toSeq // this means one or the other
     }
 
-    component.getMethods
-      .filter(hasEventSourcedEntitySubscription)
-      .sorted // make sure we get the methods in deterministic order
-      .map { method =>
-        // validate that all updates return the same type
-        val entityClass = method.getAnnotation(classOf[Subscribe.EventSourcedEntity]).value().asInstanceOf[Class[_]]
+    def validateSameType(methods: Seq[Method], tableType: Class[_]): Seq[Method] = {
 
-        previousEntityClass match {
-          case Some(`entityClass`) => // ok
-          case Some(other) =>
-            throw InvalidComponentException(
-              s"All update methods must return the same type, but [${method.getName}] returns [${entityClass.getName}] while a previous update method returns [${other.getName}]")
-          case None => previousEntityClass = Some(entityClass)
-        }
+      def invalidComponentException(method: Method, extraMessage: String) = {
+        throw InvalidComponentException(
+          s"Method [${method.getName}] annotated with '@Subscribe.EventSourcedEntity' should either receive " +
+          "a single parameter of one of the event types or " +
+          s"two ordered parameters  of type [${tableType.getName}] and an event type. $extraMessage")
+      }
```

Comment:
This is obsolete. We removed that option. Don't need to fix in this PR, I will create a ticket to clean this up.

Reply:
for your reference @octonato 
https://github.com/lightbend/kalix/issues/7733

Reply:
I had already created one. I mentioned that I was about to do it. 
https://github.com/lightbend/kalix/issues/7708

Reply:
Ok. I thought you may forget with all the spring tasks you where creating. I'll delete it.